src/main/scala/viper/gobra/backend/BackendVerifier.scala

```diff
@@ -57,23 +57,22 @@ object BackendVerifier {
 
         val programs = ChopperUtil.computeChoppedPrograms(task, pkgInfo)(config)
         val num = programs.size
+        var counter = 0 // verification progress counter
+
+        // Starts verifying all chopped programs in parallel
+        val partialVerificationResults = Future.traverse(programs.zipWithIndex) { case (program, idx) =>
+          val programID = s"${config.taskName}_$idx"
+          verifier.verify(programID, reporter, program)(executor).andThen { _ =>
+            // this block ensures that progress messages are printed in order
+            this.synchronized { counter += 1; config.reporter report ChoppedProgressMessage(counter, num) }
```

Comment:
This way you do not know which chopped program has completed, right?
I think it would be usefull to know if some chopped parts are very quick or very slow. Could you integrate the idx into the ChoppedProgressMessage?

Reply:
should be done now